modules/um_regonlymsg.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * modules/um_regonlymsg.c
+ * Copyright (c) 2020 Ariadne Conill <ariadne@dereferenced.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. The name of the author may not be used to endorse or promote products
+ *    derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "stdinc.h"
+#include "modules.h"
+#include "hook.h"
+#include "client.h"
+#include "ircd.h"
+#include "send.h"
+#include "hash.h"
+#include "s_conf.h"
+#include "s_user.h"
+#include "s_serv.h"
+#include "numeric.h"
+#include "privilege.h"
+#include "s_newconf.h"
+
+static int
+um_regonlymsg_modinit(void)
+{
+	user_modes['R'] = find_umode_slot();
+	construct_umodebuf();
+
+	return 0;
+}
+
+static void
+um_regonlymsg_moddeinit(void)
+{
+	user_modes['R'] = 0;
+	construct_umodebuf();
+}
+
+#define IsSetRegOnlyMsg(c)	((c->umodes & user_modes['R']) == user_modes['R'])
```

Comment:
Aren't these bitfields? `(foo & bit) == bit` is redundant.

Reply:
when checking for a single bit, sure.  when checking for multiple bits, no.  but yes, technically redundant i suppose.

Reply:
I prefer to keep it this way because it clarifies what the value of the bitwise-and should be.

Reply:
If you insist, then `um_regonlymsg_modinit()` needs to return `-1` if `find_umode_slot()` fails, otherwise this will always return true.

Reply:
tbh all the modules using `find_umode_slot()` should return -1.  i added that to this one.